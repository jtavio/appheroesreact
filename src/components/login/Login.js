import React, { useContext } from 'react'
import { AuthContext } from '../../auth/AuthContext'
import { types } from '../../types/types'

const Login = ({history}) => {
    //obtengo mi context y extraigo el dispatch para login
    const {dispatch} = useContext(AuthContext)

    const handleLogin = () => {
        //history.push('/')
        const lastPath = localStorage.getItem('lastPath') || '/'
        //envio el type y payload
        dispatch({
            type:types.login,
            payload:{
                name:'Jonathan'
            }
        })
        history.replace(lastPath)
    }
    return (
        <div className="container my-3">
           <h1>Login</h1> 
           <hr/>
           <button
                className="btn btn-primary"
                onClick={handleLogin}
           >
               Login

           </button>
        </div>
    )
}

export default Login;