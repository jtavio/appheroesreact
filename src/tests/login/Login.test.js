import { mount } from "enzyme";
import { AuthContext } from "../../auth/AuthContext";
import Login from "../../components/login/Login";
import { types } from "../../types/types";

describe('Prueba de login', () => {
    const history = {
        replace:jest.fn()
    }
    const contextValue = {
        dispatch: jest.fn(),
        user:{
            logged: false,
        }
    }
    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <Login history={history} />
        </AuthContext.Provider>
    );
    test('debe de mostrarse correctamente', () => {
       
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de hacer el dispatch y la navegacion', () => {
       
        const handleClick = wrapper.find('button').prop('onClick');
        handleClick();

        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type:types.login,
            payload:{name:'Jonathan'}
        });
        expect(history.replace).toHaveBeenCalledWith('/');
        localStorage.setItem('lastPath','/dc');
        handleClick();
        expect(history.replace).toHaveBeenCalledWith('/dc');
    });
    
    
    
})