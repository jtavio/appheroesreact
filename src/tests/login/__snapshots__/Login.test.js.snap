// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de login debe de mostrarse correctamente 1`] = `
<div
  className="container my-3"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
